opds.py

```diff
@@ -79,12 +79,47 @@ class Annotator(object):
 
     opds_cache_field = Work.simple_opds_entry.name
 
-    @classmethod
-    def annotate_work_entry(cls, work, active_license_pool, edition,
+    def annotate_work_entry(self, work, active_license_pool, edition,
```

Comment:
Making this an instance method instead of a class method required a lot of follow-on changes. I made this change because _sometimes_ this method needs to be an instance method, and I wasn't able to work out a way to call a method that might be a class method or an instance method depending on the subclass.

It didn't really need to be a class method, so I don't think this is a big problem.

Reply:
A lot of the additions to this class are moving the code that needs to know about a LicensePool or Identifier out of _make_entry_xml.